js/scripts.js

```diff
@@ -0,0 +1,85 @@
+
+
+var state = {
+	"one": false,
+	"two": false,
+	"three": false,
+	"four": false,
+	"five": false,
+	"six": false,
+	"seven": false,
+	"eight": false,
+	"nine": false,
+};
+document.addEventListener("DOMContentLoaded", function(){
+	document.getElementById("one").addEventListener("click", buttonListener1);
+	document.getElementById("two").addEventListener("click", buttonListener1);
+	document.getElementById("three").addEventListener("click", buttonListener1);
+	document.getElementById("four").addEventListener("click", buttonListener1);
+	document.getElementById("five").addEventListener("click", buttonListener1);
+	document.getElementById("six").addEventListener("click", buttonListener1);
+	document.getElementById("seven").addEventListener("click", buttonListener1);
+	document.getElementById("eight").addEventListener("click", buttonListener1);
+	document.getElementById("nine").addEventListener("click", buttonListener1);
+	document.getElementById("clearB").addEventListener("click", clrListener);
+});
+
+var clrListener = function() {
+	event.preventDefault();
+	for (num in state)
+	{
+		state[num] = false;
+		document.getElementById(num).className = "sq";
+		document.getElementById(num).addEventListener("click", buttonListener1);
+		document.getElementById(num).innerHTML = "?";
+	}
+}
+
+var buttonListener1 = function (e) {
+	event.preventDefault();
+	console.log("worked");
+	console.log(e);
+	console.log(e.target.id);
+	if (e.target.innerHTML ==="X")
+	{
+		e.target.innerHTML = "O";
+		e.target.className = "sqg";
+	}
+	else {
+		e.target.innerHTML = "X";
+		e.target.className = "sqb";
+	}
+	console.log("ID " + e.target.id);
+	//console.log("State " +state(e.target.id));
+	console.log(state["one"])
+	state[e.target.id] = true;
+	for(num in state)
+	{
+		if (state[num]===true && num != e.target.id)
+		{
+			document.getElementById(num).removeEventListener("click", buttonListener1);
+		}
+	}
+	
+
+	
+
+}
```

Comment:
Good attempt at the buttonListener1, you were really close! You had a lot of the right ideas, (changing the state, clearing the button listener, changing the innerHTML & class), but the function needs a bit of tweaking to get the game to work properly. Here's a revised version that will achieve the desired results. Note that I added a variable to keep track of who the current player is and change it once a move has been played.

```
var player = 'X';
var buttonListener1 = function (e) {
	if (e.target.innerHTML.indexOf('?') !== -1 && player === 'X') {
		e.target.innerHTML = "X";
		e.target.className = "sqg";
		state[e.target.id] = true;
		document.getElementById(e.target.id).removeEventListener("click", buttonListener1);
		player = 'O';
	} else if (e.target.innerHTML.indexOf('?') !== -1 && player === 'O'){
		e.target.innerHTML = "O";
		e.target.className = "sqb";
		state[e.target.id] = true;
		document.getElementById(e.target.id).removeEventListener("click", buttonListener1);
		player = 'X'
	}
}
```